### ISSUE TITLE
Incorrect reporting that Toolbar (and possibly other roles) doesn't allow aria-orientation

### ISSUE THREAD

**Issue by afercia** [dequelabs/axe-core#883]
Similar to #276 

Hello! In ARIA 1.1 the role `toolbar` supports an `aria-orientation` property with an implicit value of `horizontal`, see https://www.w3.org/TR/wai-aria-1.1/#toolbar where "implicit" doesn't prohibit to explicitly set `aria-orientation`. Regardless, given a markup like:

`<div role="toolbar" aria-orientation="horizontal" aria-label="My Toolbar">`

aXe reports a `critical violation` with the message `ARIA attribute is not allowed: aria-orientation="horizontal"`

Something has changed in ARIA 1.1 and now aria-orientation can be used on the following roles, see https://www.w3.org/TR/wai-aria-1.1/#aria-orientation
- scrollbar
- select
- separator
- slider
- tablist
- toolbar

while in ARIA 1.0 it was used only on the roles scrollbar, separator, slider. See https://www.w3.org/TR/wai-aria-1.0/states_and_properties#aria-orientation

Tested on:
Firefox 60 with  axe-devtools 3.3.0 / axe-core 3.0.0
Chrome 66 dev tools Accessibility audit

<img width="964" alt="screen shot 2018-05-10 at 09 58 42" src="https://user-images.githubusercontent.com/1682452/39859831-805e1682-543b-11e8-80ae-9ff90ac7c101.png">

<img width="728" alt="screen shot 2018-05-10 at 09 45 12" src="https://user-images.githubusercontent.com/1682452/39859833-807c307c-543b-11e8-9d4a-889994932c41.png">

**Comment by WilcoFiers** [comment]
This is being flagged because we don't think it is all that widely supported. We could test it, but relatively few ARIA 1.1 attributes are supported, so I'm not too optimistic about it.

**Comment by afercia** [comment]
@WilcoFiers thanks for your reply. Well it may or may not be supported. If not, it is ignored. Not certainly a "critical violation" though, as it is perfectly valid. Also, I don't see how this is different from #276, one could argue that aria-orientation on listbox may not be supported too.

**Comment by WilcoFiers** [comment]
Put some more thought into this. I think aria-orientation should be permitted by Axe, even when not supported I don't think the absence of it will be too much of a hinderance.

**Comment by jkodu** [comment]
@afercia

I just ran axe on your test snippet `<div role="toolbar" aria-orientation="horizontal" aria-label="My Toolbar">` and tried various other combinations of `aria-orientation` and `role`.

I do not see the violation reported. We have re-worked the `aria-allowed-role` rule and the respective checks, and this may have been fixed as a consequence of the same.

Could you validate your usecases against the latest version 3.1.1 (https://www.npmjs.com/package/axe-core) and if this is still relevant please update us accordingly and feel free to open the issue.
